src/ADAL.PCL/Internal/Telemetry/ApiEvent.cs

```diff
@@ -0,0 +1,147 @@
+        }
+
+        internal string Tenant { get; set; }
+
+        internal string Idp { get; set; }
+
+        internal string UserId { get; set; }
+
+        internal string Authority { get; set; }
+
+        internal string AuthorityType { get; set; }
+
+        internal string ValidateAuthority { get; set; }
+
+        internal string DisplayableId { get; set; }
+
+        internal string IdentityProvider { get; set; }
+
+        internal bool IsDeprecated { get; set; }
+
+        internal bool ExtendedExpires { get; set; }
+
+        internal string HashTenantIdFromAuthority(string authority)
+        {
+            Uri uri = new Uri(authority);
+            Regex regex = new Regex(uri.AbsolutePath);
+            string result = regex.Replace(authority, PlatformPlugin.CryptographyHelper.CreateSha256Hash(uri.AbsolutePath), 1);
+            return result;
+        }
+
+        internal override void SetEvent(string eventName, string eventParameter)
+        {
+            if (!string.IsNullOrEmpty(eventParameter))
+            {
+                EventDictitionary[eventName] = eventParameter;
+            }
+        }
+
+        internal void SetExtraQueryParameters(string extraQueryParameter)
+        {
+            string[] result = extraQueryParameter.Split('&');
+            StringBuilder stringBuilder = new StringBuilder();
+            foreach (string s in result)
+            {
+                if (s.Contains("="))
+                {
+                    stringBuilder.Append(s.Split('=')[0]).Append("&");
+                }
+            }
+
+            if (string.IsNullOrEmpty(stringBuilder.ToString()))
+            {
+                SetEvent(EventConstants.ExtraQueryParameters,
+                    stringBuilder.ToString().Substring(0, stringBuilder.Length - 1));
+            }
+        }
+
+        internal override void ProcessEvent(Dictionary<string, string> dispatchMap)
+        {
+            foreach (KeyValuePair<string, string> Event in EventDictitionary)
+            {
+                if (Event.Key.Equals(EventConstants.AuthorityType)
+                    || Event.Key.Equals(EventConstants.SdkVersion)
+                    || Event.Key.Equals(EventConstants.SdkPlatform)
+                    || Event.Key.Equals(EventConstants.AuthorityValidation)
+                    || Event.Key.Equals(EventConstants.ExtendedExpires)
+                    || Event.Key.Equals(EventConstants.PromptBehavior)
+                    || Event.Key.Equals(EventConstants.Idp)
+                    || Event.Key.Equals(EventConstants.Tenant)
+                    || Event.Key.Equals(EventConstants.LoginHint)
+                    || Event.Key.Equals(EventConstants.ResponseTime)
+                    || Event.Key.Equals(EventConstants.CorrelationId)
+                    || Event.Key.Equals(EventConstants.RequestId)
+                    || Event.Key.Equals(EventConstants.ApiId)
+                    || Event.Key.Equals(EventConstants.IsSuccessful)
```

Comment:
Please check line 963 of in the AuthenticationContext class.I am setting it there.